File: Directory/119. Pascal's Triangle II/Solution.py
# -*- coding: utf-8 -*-
# @Author: yufeng
# @Date:   2018-10-18 22:29:46
# @Last Modified by:   yufeng
# @Last Modified time: 2018-10-18 22:37:47
# @Email: fzhiy270@sina.com
# @Blog: http://fzhiy.com/

#  52ms beats 20.67%
#  zip(row, row) 压缩返回一个对象 zip(*)解压缩转化为列表
class Solution:
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        row = [1]
        for _ in range(rowIndex):
        	row = [x+y for x,y in zip([0]+row, row+[0])]
        return row

File: Directory/119. Pascal's Triangle II/Solution.cpp
/*
* @Author: yufeng
* @Date:   2018-10-18 22:51:28
* @Last Modified by:   yufeng
* @Last Modified time: 2018-10-18 22:55:45
* @Email: fzhiy270@sina.com
* @Blog: http://fzhiy.com/
*/

// 0ms beats 100%
class Solution {
public:
    vector<int> getRow(int rowIndex) {
    	std::vector<int> res={1};
        for (double i=1; i<=rowIndex; i++){
        	res.push_back(res.back()*(rowIndex-i+1)/i);
        }
        return res;
    }
};